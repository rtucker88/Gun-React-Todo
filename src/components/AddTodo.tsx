import * as React from "react";
import { InputGroup, Button } from "@blueprintjs/core";
import { compose, withHandlers, withStateHandlers, setDisplayName, StateHandlerMap, StateHandler } from "recompose";

import "../assets/scss/AddTodo.scss";
import { translate } from "react-i18next";
import { TransProps } from "react-i18next/src/trans";

export interface AddTodoProps {
    onAddTodo: (text: string) => void;
}

export interface AddTodoState {
    text: string;
}

export interface AddTodoStateHandlers extends StateHandlerMap<AddTodoState> {
    onInputChange: StateHandler<AddTodoState>;
    onAddNewTodo: StateHandler<AddTodoState>;
}

interface AddTodoHandlers {
    onKeyDown: (event: React.KeyboardEvent<HTMLInputElement>) => void;
}

export const stateHandlers = withStateHandlers<AddTodoState, AddTodoStateHandlers, AddTodoProps>({ text: "" }, {
    onInputChange: () => (event: React.ChangeEvent<HTMLInputElement>) => {
        return {
            text: event.target.value,
        };
    },
    onAddNewTodo: (state, props) => () => {
        props.onAddTodo(state.text);

        return {
            text: "",
        };
    },
});

export const handlers = withHandlers<AddTodoProps & AddTodoStateHandlers, AddTodoHandlers>({
    onKeyDown: (props: AddTodoCombinedProps) => (event: React.KeyboardEvent<HTMLInputElement>) => {
        if (event.keyCode === 13) {
            props.onAddNewTodo();
        }
    }
});

const enhance = compose<AddTodoCombinedProps, AddTodoProps>(setDisplayName("AddTodo"), translate("addTodos"), stateHandlers, handlers);

export type AddTodoCombinedProps = AddTodoState & AddTodoStateHandlers & AddTodoHandlers & TransProps & AddTodoProps;

export const AddTodo: React.StatelessComponent<AddTodoCombinedProps> = ({ onAddNewTodo, onKeyDown, onInputChange, t, text }) => {
    return (
        <div className="add-todo-container">
            <div className="add-todo">
                <InputGroup onKeyDown={onKeyDown} onChange={onInputChange} value={text} className="add-todo-input" placeholder={t("enterANewTodo")}/>
                <Button onClick={onAddNewTodo} icon="add" minimal={true} />
            </div>
        </div>);
};

export default enhance(AddTodo);
