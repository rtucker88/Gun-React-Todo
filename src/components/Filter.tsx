import * as React from "react";
import { ButtonGroup, Button } from "@blueprintjs/core";

import "../assets/scss/Filter.scss";
import { TransProps } from "react-i18next/src/trans";
import { translate } from "react-i18next";
import { compose, setDisplayName, withHandlers } from "recompose";

interface FilterProps {
    filterToggle: FilterToggle;
    onUpdateFilter: (toggle: FilterToggle) => void;
}

interface FilterHandlerProps {
    updateAll: () => void;
    updateIncomplete: () => void;
    updateCompleted: () => void;
}

export type FilterCombinedProps = FilterProps & TransProps & FilterHandlerProps;

export const handlers = withHandlers<FilterProps, FilterHandlerProps>({
    updateAll: props => () => props.onUpdateFilter("ALL"),
    updateIncomplete: props => () => props.onUpdateFilter("INCOMPLETE"),
    updateCompleted: props => () => props.onUpdateFilter("COMPLETED"),
});

const enhance = compose<{}, FilterProps>(setDisplayName("Filter"), translate("filterToggle"), handlers);

export type FilterToggle = "ALL" | "COMPLETED" | "INCOMPLETE";

const Filter: React.SFC<FilterCombinedProps> = ({ filterToggle, updateAll, updateCompleted, updateIncomplete, t }) => {
    return (
    <div className="filter">
        <ButtonGroup>
            <Button active={filterToggle === "ALL"} onClick={updateAll}>{t("all")}</Button>
            <Button active={filterToggle === "INCOMPLETE"} onClick={updateIncomplete} icon="square">{t("incomplete")}</Button>
            <Button active={filterToggle === "COMPLETED"} onClick={updateCompleted} icon="tick">{t("completed")}</Button>
        </ButtonGroup>
    </div>);
};

export default enhance(Filter);