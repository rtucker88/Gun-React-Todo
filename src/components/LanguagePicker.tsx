import * as React from "react";
import { Button, Popover, MenuItem, Menu, Position } from "@blueprintjs/core";
import UnreachableCaseError from "../utils/UnreachableErrorCase";
import { setDisplayName, pure, compose } from "recompose";

interface LanguagePickerProps {
    onChangeLanguage: (language: I18nLanguage) => void;
    currentLanguage: I18nLanguage;
}

export type I18nLanguage = "en" | "de" | "el" | "uk";

interface MenuItemConfig {
    country: string;
    languageName: string;
    onClickParameter: I18nLanguage;
}

const menuItems: MenuItemConfig[] = [{
    country: "US",
    languageName: "English",
    onClickParameter: "en"
},
{
    country: "DE",
    languageName: "Deutsch",
    onClickParameter: "de"
},
{
    country: "GR",
    languageName: "ελληνικά",
    onClickParameter: "el"
},
{
    country: "UA",
    languageName: "Українська",
    onClickParameter: "uk"
}];

const LanguagePicker: React.StatelessComponent<LanguagePickerProps> = ({ currentLanguage, onChangeLanguage }) => {
    return (
        <Popover content={getMenuItems(onChangeLanguage)} position={Position.BOTTOM_LEFT}>
            <Button icon="caret-down" text={getCurrentLocaleFlag(currentLanguage)} minimal={true}></Button>
        </Popover>
    );
};

const enhance = compose<{}, LanguagePickerProps>(setDisplayName("LanguagePicker"), pure);
export default enhance(LanguagePicker);

export function getMenuItems(onClick: (language: I18nLanguage) => void) {
    return (<Menu>
        {menuItems.map(({ country, languageName, onClickParameter }) => {
            return (<MenuItem key={country} text={`${getFlagFromCountryCode(country)} ${languageName}`} onClick={() => onClick(onClickParameter)} />);
        })}
    </Menu>);
}


export function getCurrentLocaleFlag(locale: I18nLanguage): string {
    switch (locale) {
        case "en":
            return getFlagFromCountryCode("US");
        case "de":
            return getFlagFromCountryCode("DE");
        case "el":
            return getFlagFromCountryCode("GR");
        case "uk":
            return getFlagFromCountryCode("UA");
        default:
            throw new UnreachableCaseError(locale);
    }
}

export function getFlagFromCountryCode(countryCode: string): string {
    // Taken from https://binarypassion.net/lets-turn-an-iso-country-code-into-a-unicode-emoji-shall-we-870c16e05aad
    return countryCode.toUpperCase().replace(/./g, char => String.fromCodePoint(char.charCodeAt(0) + 127397));
}