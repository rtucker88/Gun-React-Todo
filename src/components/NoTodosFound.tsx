import * as React from "react";
import { NonIdealState } from "@blueprintjs/core";

import "../assets/scss/NoTodosFound.scss";
import { translate } from "react-i18next";
import { TransProps } from "react-i18next/src/trans";
import { setDisplayName, compose, pure } from "recompose";

interface NoTodosFoundProps {}

type NoTodosFoundCombinedProps = NoTodosFoundProps & TransProps;

const NoTodosFound: React.SFC<NoTodosFoundCombinedProps> = ({ t }) => {
    return (<div className="no-todos-found">
        <NonIdealState title={t("noneFoundTitle")} description={t("noneFoundDescription")} icon="zoom-out" />
    </div>);
};

const enhance = compose(setDisplayName("NoTodosFound"), pure, translate("noTodosFound"));

export default enhance(NoTodosFound);
