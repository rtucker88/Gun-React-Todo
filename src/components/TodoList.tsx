import * as React from "react";
import TodoItem from "./Todo";

import "../assets/scss/TodoList.scss";
import { compose, setDisplayName, pure } from "recompose";

export interface Todo {
    contents: string;
    completed: boolean;
    timeCreated: number;
    id: string;
}

interface TodoListProps {
    todos: Todo[];
    onDelete: (id: string) => void;
    onComplete: (id: string) => void;
}

const TodoList: React.SFC<TodoListProps> = ({ onComplete, onDelete, todos }) => {
    return (<div className="todo-list">
        {todos.map(({ completed, contents, id, timeCreated }) => <TodoItem completed={completed} contents={contents} key={`${timeCreated}-${contents}`} onDelete={onDelete} onComplete={onComplete} id={id} timeCreated={timeCreated} />)}
    </div>);
};

const enhance = compose<{}, TodoListProps>(setDisplayName("TodoList"), pure);

export default enhance(TodoList);
