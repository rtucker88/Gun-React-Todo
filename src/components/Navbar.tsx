import * as React from "react";
import { Navbar as BlueprintNavbar, NavbarGroup, NavbarHeading, NavbarDivider, Button, Classes, Alignment } from "@blueprintjs/core";
import { default as LanguagePicker, I18nLanguage } from "./LanguagePicker";
import { translate } from "react-i18next";
import { TransProps } from "react-i18next/src/trans";
import { compose, setDisplayName, pure } from "recompose";

interface NavbarProps {
    onChangeLanguage: (language: I18nLanguage) => void;
    currentLanguage: I18nLanguage;
}

type CombinedNavbarProps = NavbarProps & TransProps;

const Navbar: React.SFC<CombinedNavbarProps> = ({ onChangeLanguage, currentLanguage, t }) => {
    return (<BlueprintNavbar>
            <NavbarGroup>
                <NavbarHeading>Gun-React-Todo</NavbarHeading>
                <NavbarDivider />
                <Button className={Classes.MINIMAL} icon="home" text={t("home")} />
                <Button className={Classes.MINIMAL} icon="help" text={t("about")} />
            </NavbarGroup>
            <NavbarGroup align={Alignment.RIGHT}>
                <LanguagePicker onChangeLanguage={onChangeLanguage} currentLanguage={currentLanguage} />
            </NavbarGroup>
        </BlueprintNavbar>);
};

const enhance = compose<{}, NavbarProps>(setDisplayName("Navbar"), pure, translate("navbar"));

export default enhance(Navbar);