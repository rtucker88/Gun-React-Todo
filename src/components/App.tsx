import * as React from "react";
import * as uuidv4 from "uuid/v4";
import * as i18nnext from "i18next";
import Navbar from "./Navbar";
import { TodoList, Todo } from "./TodoList";
import AddTodo from "./AddTodo";
import { default as Filter, FilterToggle } from "./Filter";
import UnreachableCaseError from "../utils/UnreachableErrorCase";
import NoTodosFound from "./NoTodosFound";

import "../assets/scss/App.scss";
import { I18nLanguage } from "./LanguagePicker";
import i18n from "../i18n";

export interface AppProps {
}

interface AppState {
    todos: Todo[];
    filterToggle: FilterToggle;
    currentLanguage: I18nLanguage;
}

export default class App extends React.Component<AppProps, AppState> {
    state = {
        todos: [],
        filterToggle: "ALL" as FilterToggle,
        currentLanguage: "en" as I18nLanguage,
    };

    componentWillMount() {
        // Trim this here because we'll get things like en-US and en-UK
        this.setState({
            currentLanguage: (i18nnext as any).default.language.substr(0, 2) as I18nLanguage,
        });
    }

    private static getFilteredTodos(todos: Todo[], toggle: FilterToggle): Todo[] {
        return todos.filter(todo => {
            switch (toggle) {
                case "ALL":
                    return true;
                case "COMPLETED":
                    return todo.completed;
                case "INCOMPLETE":
                    return !todo.completed;
                default:
                    throw new UnreachableCaseError(toggle);
            }
        });
    }

    private handleAddTodo = (text: string) => {
        this.setState((state: Readonly<AppState>) => {
            return {
                todos: [...state.todos, { contents: text, completed: false, timeCreated: Date.now(), id: uuidv4() }],
            };
        });
    }

    private handleTodoCompletion = (id: string) => {
        this.setState((state: Readonly<AppState>) => {
            const todo = state.todos.find(todo => todo.id === id);
            todo.completed = !todo.completed;

            return {
                todos: state.todos,
            };
        });
    }

    private handleTodoDelete = (id: string) => {
        this.setState((state: Readonly<AppState>) => {
            const todoIndex = state.todos.findIndex(todo => todo.id === id);
            state.todos.splice(todoIndex, 1);

            return {
                todos: state.todos,
            };
        });
    }

    private handleUpdateFilter = (filterToggle: FilterToggle) => {
        this.setState({
            filterToggle,
        });
    }

    private handleLanguageChange = (currentLanguage: I18nLanguage) => {
        i18n.changeLanguage(currentLanguage);

        this.setState({
            currentLanguage,
        });
    }

    render() {
        const { currentLanguage, filterToggle, todos } = this.state;

        const filteredTodos = App.getFilteredTodos(todos, filterToggle);

        return (
            <div className="app">
                <Navbar currentLanguage={currentLanguage} onChangeLanguage={this.handleLanguageChange}></Navbar>
                <h1 className="todo-header">Todos</h1>
                <AddTodo onAddTodo={this.handleAddTodo} />
                <Filter filterToggle={filterToggle} onUpdateFilter={this.handleUpdateFilter} />
                {filteredTodos.length ? <TodoList todos={filteredTodos} onComplete={this.handleTodoCompletion} onDelete={this.handleTodoDelete} /> : <NoTodosFound />}
            </div>
        );
    }
}
