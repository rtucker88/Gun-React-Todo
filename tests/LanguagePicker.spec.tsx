import * as React from "react";
import { mount } from "enzyme";
import { I18nLanguage, default as LanguagePicker, getFlagFromCountryCode, getCurrentLocaleFlag, getMenuItems } from "../src/components/LanguagePicker";
import { MenuItem } from "@blueprintjs/core";

describe("LanguagePicker", () => {
    it("renders correctly", () => {
        const languages: I18nLanguage[] = ["en", "de", "el", "uk"];

        languages.forEach(language => {
            const wrapper = mount(<LanguagePicker currentLanguage={language} onChangeLanguage={jest.fn()} />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    it("gets the correct flag from the country code", () => {
        const brazil = "BR";
        const flag = getFlagFromCountryCode(brazil);

        expect(flag).toEqual("ðŸ‡§ðŸ‡·");
    });

    it("gets the current locale flag", () => {
        const languages: I18nLanguage[] = ["en", "de", "el", "uk"];
        const flags = ["ðŸ‡ºðŸ‡¸", "ðŸ‡©ðŸ‡ª", "ðŸ‡¬ðŸ‡·", "ðŸ‡ºðŸ‡¦"];

        languages.forEach((language, idx) => {
            const flag = getCurrentLocaleFlag(language);
            expect(flag).toEqual(flags[idx]);
        });

        expect(() => {
            getCurrentLocaleFlag("XYZ" as I18nLanguage);
        }).toThrow();
    });

    it("gets the menu items", () => {
        const menuItems = getMenuItems(jest.fn());
        const wrapper = mount(menuItems);
        expect(wrapper).toMatchSnapshot();
    });

    it("handles a click on a menu item", () => {
        const mockClickHandler = jest.fn();
        const menuItems = getMenuItems(mockClickHandler);

        const wrapper = mount(menuItems);
        const englishMenuItem = wrapper.find(MenuItem).at(0);
        englishMenuItem.prop("onClick")({} as React.MouseEvent<HTMLAnchorElement>);
        expect(mockClickHandler.mock.calls[0][0]).toEqual("en");
    });
});
