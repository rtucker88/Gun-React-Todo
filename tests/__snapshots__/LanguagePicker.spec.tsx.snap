// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguagePicker gets the menu items 1`] = `
<Blueprint3.Menu>
  <ul
    className="bp3-menu"
  >
    <Blueprint3.MenuItem
      disabled={false}
      key="US"
      multiline={false}
      onClick={[Function]}
      popoverProps={Object {}}
      shouldDismissPopover={true}
      text="🇺🇸 English"
    >
      <li
        className=""
      >
        <a
          className="bp3-menu-item bp3-popover-dismiss"
          onClick={[Function]}
        >
          <Blueprint3.Icon />
          <Blueprint3.Text
            className="bp3-fill"
            ellipsize={true}
          >
            <div
              className="bp3-text-overflow-ellipsis bp3-fill"
            >
              🇺🇸 English
            </div>
          </Blueprint3.Text>
        </a>
      </li>
    </Blueprint3.MenuItem>
    <Blueprint3.MenuItem
      disabled={false}
      key="DE"
      multiline={false}
      onClick={[Function]}
      popoverProps={Object {}}
      shouldDismissPopover={true}
      text="🇩🇪 Deutsch"
    >
      <li
        className=""
      >
        <a
          className="bp3-menu-item bp3-popover-dismiss"
          onClick={[Function]}
        >
          <Blueprint3.Icon />
          <Blueprint3.Text
            className="bp3-fill"
            ellipsize={true}
          >
            <div
              className="bp3-text-overflow-ellipsis bp3-fill"
            >
              🇩🇪 Deutsch
            </div>
          </Blueprint3.Text>
        </a>
      </li>
    </Blueprint3.MenuItem>
    <Blueprint3.MenuItem
      disabled={false}
      key="GR"
      multiline={false}
      onClick={[Function]}
      popoverProps={Object {}}
      shouldDismissPopover={true}
      text="🇬🇷 ελληνικά"
    >
      <li
        className=""
      >
        <a
          className="bp3-menu-item bp3-popover-dismiss"
          onClick={[Function]}
        >
          <Blueprint3.Icon />
          <Blueprint3.Text
            className="bp3-fill"
            ellipsize={true}
          >
            <div
              className="bp3-text-overflow-ellipsis bp3-fill"
            >
              🇬🇷 ελληνικά
            </div>
          </Blueprint3.Text>
        </a>
      </li>
    </Blueprint3.MenuItem>
    <Blueprint3.MenuItem
      disabled={false}
      key="UA"
      multiline={false}
      onClick={[Function]}
      popoverProps={Object {}}
      shouldDismissPopover={true}
      text="🇺🇦 Українська"
    >
      <li
        className=""
      >
        <a
          className="bp3-menu-item bp3-popover-dismiss"
          onClick={[Function]}
        >
          <Blueprint3.Icon />
          <Blueprint3.Text
            className="bp3-fill"
            ellipsize={true}
          >
            <div
              className="bp3-text-overflow-ellipsis bp3-fill"
            >
              🇺🇦 Українська
            </div>
          </Blueprint3.Text>
        </a>
      </li>
    </Blueprint3.MenuItem>
  </ul>
</Blueprint3.Menu>
`;

exports[`LanguagePicker renders correctly 1`] = `
<LanguagePicker
  currentLanguage="en"
  onChangeLanguage={[MockFunction]}
>
  <LanguagePicker
    currentLanguage="en"
    onChangeLanguage={[MockFunction]}
  >
    <Blueprint3.Popover
      captureDismiss={false}
      content={
        <Blueprint3.Menu>
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇺🇸 English"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇩🇪 Deutsch"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇬🇷 ελληνικά"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇺🇦 Українська"
          />
        </Blueprint3.Menu>
      }
      defaultIsOpen={false}
      disabled={false}
      hasBackdrop={false}
      hoverCloseDelay={300}
      hoverOpenDelay={150}
      inheritDarkTheme={true}
      interactionKind="click"
      minimal={false}
      modifiers={Object {}}
      openOnTargetFocus={true}
      position="bottom-left"
      targetTagName="span"
      transitionDuration={300}
      usePortal={true}
      wrapperTagName="span"
    >
      <Manager>
        <span
          className="bp3-popover-wrapper"
        >
          <Reference
            innerRef={[Function]}
          >
            <InnerReference
              getReferenceRef={[Function]}
              innerRef={[Function]}
            >
              <Blueprint3.ResizeSensor
                onResize={[Function]}
              >
                <span
                  className="bp3-popover-target"
                  onClick={[Function]}
                >
                  <Blueprint3.Button
                    className=""
                    icon="caret-down"
                    key=".0"
                    minimal={true}
                    text="🇺🇸"
                  >
                    <button
                      className="bp3-button bp3-minimal"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      type="button"
                    >
                      <Blueprint3.Icon
                        icon="caret-down"
                        key="leftIcon"
                      >
                        <span
                          className="bp3-icon bp3-icon-caret-down"
                          icon="caret-down"
                        >
                          <svg
                            data-icon="caret-down"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              caret-down
                            </desc>
                            <path
                              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 0 0-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                      <span
                        className="bp3-button-text"
                        key="text"
                      >
                        🇺🇸
                      </span>
                      <Blueprint3.Icon
                        key="rightIcon"
                      />
                    </button>
                  </Blueprint3.Button>
                </span>
              </Blueprint3.ResizeSensor>
            </InnerReference>
          </Reference>
          <Blueprint3.Overlay
            autoFocus={true}
            backdropClassName="bp3-popover-backdrop"
            backdropProps={Object {}}
            canEscapeKeyClose={true}
            canOutsideClickClose={true}
            enforceFocus={true}
            hasBackdrop={false}
            isOpen={false}
            lazy={true}
            onClose={[Function]}
            transitionDuration={300}
            transitionName="bp3-popover"
            usePortal={true}
          />
        </span>
      </Manager>
    </Blueprint3.Popover>
  </LanguagePicker>
</LanguagePicker>
`;

exports[`LanguagePicker renders correctly 2`] = `
<LanguagePicker
  currentLanguage="de"
  onChangeLanguage={[MockFunction]}
>
  <LanguagePicker
    currentLanguage="de"
    onChangeLanguage={[MockFunction]}
  >
    <Blueprint3.Popover
      captureDismiss={false}
      content={
        <Blueprint3.Menu>
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇺🇸 English"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇩🇪 Deutsch"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇬🇷 ελληνικά"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇺🇦 Українська"
          />
        </Blueprint3.Menu>
      }
      defaultIsOpen={false}
      disabled={false}
      hasBackdrop={false}
      hoverCloseDelay={300}
      hoverOpenDelay={150}
      inheritDarkTheme={true}
      interactionKind="click"
      minimal={false}
      modifiers={Object {}}
      openOnTargetFocus={true}
      position="bottom-left"
      targetTagName="span"
      transitionDuration={300}
      usePortal={true}
      wrapperTagName="span"
    >
      <Manager>
        <span
          className="bp3-popover-wrapper"
        >
          <Reference
            innerRef={[Function]}
          >
            <InnerReference
              getReferenceRef={[Function]}
              innerRef={[Function]}
            >
              <Blueprint3.ResizeSensor
                onResize={[Function]}
              >
                <span
                  className="bp3-popover-target"
                  onClick={[Function]}
                >
                  <Blueprint3.Button
                    className=""
                    icon="caret-down"
                    key=".0"
                    minimal={true}
                    text="🇩🇪"
                  >
                    <button
                      className="bp3-button bp3-minimal"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      type="button"
                    >
                      <Blueprint3.Icon
                        icon="caret-down"
                        key="leftIcon"
                      >
                        <span
                          className="bp3-icon bp3-icon-caret-down"
                          icon="caret-down"
                        >
                          <svg
                            data-icon="caret-down"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              caret-down
                            </desc>
                            <path
                              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 0 0-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                      <span
                        className="bp3-button-text"
                        key="text"
                      >
                        🇩🇪
                      </span>
                      <Blueprint3.Icon
                        key="rightIcon"
                      />
                    </button>
                  </Blueprint3.Button>
                </span>
              </Blueprint3.ResizeSensor>
            </InnerReference>
          </Reference>
          <Blueprint3.Overlay
            autoFocus={true}
            backdropClassName="bp3-popover-backdrop"
            backdropProps={Object {}}
            canEscapeKeyClose={true}
            canOutsideClickClose={true}
            enforceFocus={true}
            hasBackdrop={false}
            isOpen={false}
            lazy={true}
            onClose={[Function]}
            transitionDuration={300}
            transitionName="bp3-popover"
            usePortal={true}
          />
        </span>
      </Manager>
    </Blueprint3.Popover>
  </LanguagePicker>
</LanguagePicker>
`;

exports[`LanguagePicker renders correctly 3`] = `
<LanguagePicker
  currentLanguage="el"
  onChangeLanguage={[MockFunction]}
>
  <LanguagePicker
    currentLanguage="el"
    onChangeLanguage={[MockFunction]}
  >
    <Blueprint3.Popover
      captureDismiss={false}
      content={
        <Blueprint3.Menu>
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇺🇸 English"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇩🇪 Deutsch"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇬🇷 ελληνικά"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇺🇦 Українська"
          />
        </Blueprint3.Menu>
      }
      defaultIsOpen={false}
      disabled={false}
      hasBackdrop={false}
      hoverCloseDelay={300}
      hoverOpenDelay={150}
      inheritDarkTheme={true}
      interactionKind="click"
      minimal={false}
      modifiers={Object {}}
      openOnTargetFocus={true}
      position="bottom-left"
      targetTagName="span"
      transitionDuration={300}
      usePortal={true}
      wrapperTagName="span"
    >
      <Manager>
        <span
          className="bp3-popover-wrapper"
        >
          <Reference
            innerRef={[Function]}
          >
            <InnerReference
              getReferenceRef={[Function]}
              innerRef={[Function]}
            >
              <Blueprint3.ResizeSensor
                onResize={[Function]}
              >
                <span
                  className="bp3-popover-target"
                  onClick={[Function]}
                >
                  <Blueprint3.Button
                    className=""
                    icon="caret-down"
                    key=".0"
                    minimal={true}
                    text="🇬🇷"
                  >
                    <button
                      className="bp3-button bp3-minimal"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      type="button"
                    >
                      <Blueprint3.Icon
                        icon="caret-down"
                        key="leftIcon"
                      >
                        <span
                          className="bp3-icon bp3-icon-caret-down"
                          icon="caret-down"
                        >
                          <svg
                            data-icon="caret-down"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              caret-down
                            </desc>
                            <path
                              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 0 0-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                      <span
                        className="bp3-button-text"
                        key="text"
                      >
                        🇬🇷
                      </span>
                      <Blueprint3.Icon
                        key="rightIcon"
                      />
                    </button>
                  </Blueprint3.Button>
                </span>
              </Blueprint3.ResizeSensor>
            </InnerReference>
          </Reference>
          <Blueprint3.Overlay
            autoFocus={true}
            backdropClassName="bp3-popover-backdrop"
            backdropProps={Object {}}
            canEscapeKeyClose={true}
            canOutsideClickClose={true}
            enforceFocus={true}
            hasBackdrop={false}
            isOpen={false}
            lazy={true}
            onClose={[Function]}
            transitionDuration={300}
            transitionName="bp3-popover"
            usePortal={true}
          />
        </span>
      </Manager>
    </Blueprint3.Popover>
  </LanguagePicker>
</LanguagePicker>
`;

exports[`LanguagePicker renders correctly 4`] = `
<LanguagePicker
  currentLanguage="uk"
  onChangeLanguage={[MockFunction]}
>
  <LanguagePicker
    currentLanguage="uk"
    onChangeLanguage={[MockFunction]}
  >
    <Blueprint3.Popover
      captureDismiss={false}
      content={
        <Blueprint3.Menu>
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇺🇸 English"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇩🇪 Deutsch"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇬🇷 ελληνικά"
          />
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            onClick={[Function]}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="🇺🇦 Українська"
          />
        </Blueprint3.Menu>
      }
      defaultIsOpen={false}
      disabled={false}
      hasBackdrop={false}
      hoverCloseDelay={300}
      hoverOpenDelay={150}
      inheritDarkTheme={true}
      interactionKind="click"
      minimal={false}
      modifiers={Object {}}
      openOnTargetFocus={true}
      position="bottom-left"
      targetTagName="span"
      transitionDuration={300}
      usePortal={true}
      wrapperTagName="span"
    >
      <Manager>
        <span
          className="bp3-popover-wrapper"
        >
          <Reference
            innerRef={[Function]}
          >
            <InnerReference
              getReferenceRef={[Function]}
              innerRef={[Function]}
            >
              <Blueprint3.ResizeSensor
                onResize={[Function]}
              >
                <span
                  className="bp3-popover-target"
                  onClick={[Function]}
                >
                  <Blueprint3.Button
                    className=""
                    icon="caret-down"
                    key=".0"
                    minimal={true}
                    text="🇺🇦"
                  >
                    <button
                      className="bp3-button bp3-minimal"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      type="button"
                    >
                      <Blueprint3.Icon
                        icon="caret-down"
                        key="leftIcon"
                      >
                        <span
                          className="bp3-icon bp3-icon-caret-down"
                          icon="caret-down"
                        >
                          <svg
                            data-icon="caret-down"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              caret-down
                            </desc>
                            <path
                              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 0 0-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                      <span
                        className="bp3-button-text"
                        key="text"
                      >
                        🇺🇦
                      </span>
                      <Blueprint3.Icon
                        key="rightIcon"
                      />
                    </button>
                  </Blueprint3.Button>
                </span>
              </Blueprint3.ResizeSensor>
            </InnerReference>
          </Reference>
          <Blueprint3.Overlay
            autoFocus={true}
            backdropClassName="bp3-popover-backdrop"
            backdropProps={Object {}}
            canEscapeKeyClose={true}
            canOutsideClickClose={true}
            enforceFocus={true}
            hasBackdrop={false}
            isOpen={false}
            lazy={true}
            onClose={[Function]}
            transitionDuration={300}
            transitionName="bp3-popover"
            usePortal={true}
          />
        </span>
      </Manager>
    </Blueprint3.Popover>
  </LanguagePicker>
</LanguagePicker>
`;
